src/modules/70-pipeline/components/ExecutionCompareYamls/ExecutionCompareContext.tsx

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2022 Harness Inc. All rights reserved.
+ * Use of this source code is governed by the PolyForm Shield 1.0.0 license
+ * that can be found in the licenses directory at the root of this repository, also available at
+ * https://polyformproject.org/wp-content/uploads/2020/06/PolyForm-Shield-1.0.0.txt.
+ */
+
+import React, { createContext, ReactNode, useContext, ReactElement, useState, useCallback } from 'react'
+import type { PipelineExecutionSummary } from 'services/pipeline-ng'
+
+interface ExecutionCompareContext {
+  compareItems: PipelineExecutionSummary[]
+  addToCompare: (execution: PipelineExecutionSummary) => void
+  removeFromCompare: (execution: PipelineExecutionSummary) => void
+  isCompareMode: boolean
+  cancelCompareMode: () => void
+}
+export const ExecutionCompareContext = createContext<ExecutionCompareContext>({} as ExecutionCompareContext)
+export const useExecutionCompareContext = () => useContext(ExecutionCompareContext)
+
+export function ExecutionCompareProvider({ children }: { children: ReactNode }): ReactElement {
```

Comment:
You can simplify this by using the `FC` type.
```suggestion
export const ExecutionCompareProvider: FC = ({ children }) => {
```

Reply:
I like FC, but I see we have a eslint rule that restricts FC arrow components

Reply:
FC latest removed children type, we have to type it explicitely